src/library_syscall.js

```diff
@@ -871,8 +871,9 @@ var SyscallsLibrary = {
     var buf = SYSCALLS.get(), size = SYSCALLS.get();
     if (size === 0) return -ERRNO_CODES.EINVAL;
     var cwd = FS.cwd();
-    if (size < cwd.length + 1) return -ERRNO_CODES.ERANGE;
-    writeAsciiToMemory(cwd, buf);
+    var cwdLengthInBytes = lengthBytesUTF8(cwd);
+    if (size < cwdLengthInBytes + 1) return -ERRNO_CODES.ERANGE;
+    stringToUTF8(cwd, buf, cwdLengthInBytes + 1);
```

Comment:
I think there is an issue with this line. Shouldn't it be `stringToUTF8(cwd, buf, size);`? I wonder why this was not caught in the test?

Reply:
Ah, you are definitely right. Though it would still be good to do this change, to explicitly use the size that the buffer actually is to be clear.